crates/noirc_evaluator/src/ssa_refactor/ir/dom.rs

```diff
@@ -0,0 +1,443 @@
+use std::{cmp::Ordering, collections::HashMap};
```

Comment:
same here for dominator tree -- ie why it is needed and other useful information that would help someone understand the codebase. See at the top of some of Cranelifts files for inspiration on how this is done